bf: ARSN-402 use local RequestLogger in batchDelete

jonathan-gramain opened the pull request:
**Note for reviews**: easier to review individual commits.

- bump werelogs dependency

  + typescript fixes to be compatible with the latest werelogs

- use local RequestLogger in DataWrapper.batchDelete

  Create a local RequestLogger in batchDelete(): this allows to track the elapsed time of the batch delete sub-request, and avoids being forced to create a new request logger before calling the function (due to the call to `log.end()`), which was error-prone and hardly maintainable.

- sanitize use of log object in DataWrapper.delete()

  Don't assume that we can safely call `end()` on the passed log object if there is no callback (separation of concerns). additionally, an error object was passed where `end()` expects a string as a message, causing implicit conversion.

  Since errors are already logged, there is no need to bind the `callback` object to `log.end` (there is no strong reason to log the elapsed time there, the only use I can see where we don't pass a callback in Cloudserver is to support deletion of old metadata with a string as location array. IMHO not worth the complexity of adding it there, as the rest of the API doesn't log elapsed time anyways except for `batchDelete`).

jonathan-gramain committed:
ARSN-402 bump werelogs dependency

+ typescript fixes to be compatible with the latest werelogs

jonathan-gramain committed:
bf: ARSN-402 use local RequestLogger in batchDelete

Create a local RequestLogger in batchDelete(): this allows to track
the elapsed time of the batch delete sub-request, and avoids being
forced to create a new request logger before calling the function (due
to the call to `log.end()`), which was error-prone and hardly
maintainable.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/arsenal/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ARSN-402 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.61`

* `7.70.27`

* `8.1.125`


Please check the `Fix Version/s` of ARSN-402, or the target
branch of this pull request.

jonathan-gramain commented:
ping

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

jonathan-gramain commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger` with contents from `w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger origin/development/8.1
 $ git merge origin/w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

jonathan-gramain committed:
bf: ARSN-402 sanitize use of log object in DataWrapper.delete()

Don't assume that we can safely call `end()` on the passed log object
if there is no callback (separation of concerns). Additionally, an
error object was passed where `end()` expects a string as a message,
causing implicit conversion.

Since errors are already logged, there is no need to bind the
`callback` object to `log.end` (there is no strong reason to log the
elapsed time there, the only use I can see where we don't pass a
callback in Cloudserver is to support deletion of old metadata with a
string as location array. IMHO not worth the complexity of adding it
there, as the rest of the API doesn't log elapsed time anyways except
for `batchDelete`).

jonathan-gramain committed:
ARSN-402 bump arsenal version

bert-e commented:
# History mismatch



Merge commit #4eb70ca964dfed8e718e8ce267caf6f19bbea2c4 on the integration branch
`w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` is merging a branch which is neither the current
branch `bugfix/ARSN-402-batchDeleteRequestLogger` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #4eb70ca964dfed8e718e8ce267caf6f19bbea2c4 on the integration branch
`w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` is merging a branch which is neither the current
branch `bugfix/ARSN-402-batchDeleteRequestLogger` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` with contents from `bugfix/ARSN-402-batchDeleteRequestLogger`
and `development/7.70`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger origin/development/7.70
 $ git merge origin/bugfix/ARSN-402-batchDeleteRequestLogger
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger` with contents from `w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger origin/development/8.1
 $ git merge origin/w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ARSN-402-batchDeleteRequestLogger
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




There is a conflict between your branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
destination branch `development/7.10`.

Please resolve the conflict on **the feature branch** (`bugfix/ARSN-402-batchDeleteRequestLogger`).

```sh
git fetch && \
git checkout origin/bugfix/ARSN-402-batchDeleteRequestLogger && \
git merge origin/development/7.10
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:bugfix/ARSN-402-batchDeleteRequestLogger
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #03d03793589f8bff0c91e409be8b04716795b0e8 on the integration branch
`w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` is merging a branch which is neither the current
branch `bugfix/ARSN-402-batchDeleteRequestLogger` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/approve

bert-e commented:
# History mismatch



Merge commit #03d03793589f8bff0c91e409be8b04716795b0e8 on the integration branch
`w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` is merging a branch which is neither the current
branch `bugfix/ARSN-402-batchDeleteRequestLogger` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #03d03793589f8bff0c91e409be8b04716795b0e8 on the integration branch
`w/7.70/bugfix/ARSN-402-batchDeleteRequestLogger` is merging a branch which is neither the current
branch `bugfix/ARSN-402-batchDeleteRequestLogger` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `bugfix/ARSN-402-batchDeleteRequestLogger` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.4`



Please check the status of the associated issue ARSN-402.

Goodbye jonathan-gramain.
